### PR TITLE
CSL-24: Company Registration Certificate upload

### PR DESCRIPTION
**Pull request by jamiecarterHO**
## What? 

[CSL-24](https://collaboration.homeoffice.gov.uk/jira/browse/CSL-24) Adds the configuration and files required to display the "Upload company registration certificate" page and enable uploads via file-vault to S3.

## Why?

The updates will allow deployment of a file-vault container in the namespace that can connect to the S3 bucket, and add the page and functionality for uploads

## Check list

- [x] I have reviewed my own pull request
- [ ] I have written tests (if relevant)

### CODE REVIEW

**Comment by jamiecarterHO** [comment]
Ready for first review. Some discussion to be had about where reusable methods/behaviours should be stored. I believe they can be shared cross-forms as they are but happy to entertain any alternative ideas @dk4g

File: apps/precursor-chemicals/translations/src/en/buttons.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
   "confirm-and-submit": "Confirm and submit",
   "new-application": "Submit another application",
   "choose-file": "Choose file",
-  "change": "Change", 
+  "change": "Change",
   "remove": "Remove",
-  "add-another-substance": "Add another substance"
-} 
+  "add-another-substance": "Add another substance",
+  "delete": "Remove"
```
Comment (dk4g) [inline apps/precursor-chemicals/translations/src/en/buttons.json:16]: I noticed there is a `remove` translation on line 14, which probably can be reused? but will require updates on template and js event handlers

File: apps/precursor-chemicals/views/upload-company-certificate.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    </p>
+    <input class="govuk-file-upload" id="file-upload" name="file-upload" type="file" value="company-registration-certificate">
+    <div id="upload-page-loading-spinner" class="spinner-container">
+      <div class="spinner-loader"></div>
+      <span class="spinner-message">{{#t}}journey.uploading{{/t}}</span>
+    </div>
+  </div>
+
+  {{^values.company-registration-certificate}}
+  <h2 class="govuk-heading-s">No documents uploaded</h2>
```
Comment (RobertMcCann) [inline apps/precursor-chemicals/views/upload-company-certificate.html:25]: This heading should be 'No files uploaded' to match figma

File: apps/precursor-chemicals/views/upload-company-certificate.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  {{/values.company-registration-certificate}}
+
+  {{#values.company-registration-certificate.length}}
+  <h2 class="govuk-heading-s">You have uploaded {{values.company-registration-certificate.length}} file(s)</h2>
```
Comment (RobertMcCann) [inline apps/precursor-chemicals/views/upload-company-certificate.html:29]: Think this heading should be 'Files uploaded'

Comment (dk4g) [reply apps/precursor-chemicals/views/upload-company-certificate.html:29]: Also, it's worth moving this to translations

**Comment by RobertMcCann** [comment]
We could also add the 'Evidence' summary data section and the company registration certificate field so it appears on summary page

File: apps/precursor-chemicals/translations/src/en/fields.json [hunk 4]
```diff
@@ -152,5 +152,8 @@
   },
   "invoicing-purchase-order-number": {
     "label": "Purchase order number (optional)"
+  },
+  "upload-company-certificate": {
```
Comment (dk4g) [inline apps/precursor-chemicals/translations/src/en/fields.json:156]: I would suggest to use the generic name for this field, so it can be reused

Comment (dk4g) [reply apps/precursor-chemicals/translations/src/en/fields.json:156]: withdrawn

**Review by dk4g — APPROVED** [review]

**Review by dk4g — APPROVED** [review]

**Review by RobertMcCann — APPROVED** [review]
Looks good to me

**Pull request merged**